### DISCUSSION TITLE
Make core of package more abstract and provide multiple opinionated solutions

### DISCUSSION THREAD

**Discussion by wesobi** [go-oauth2/oauth2#262] [Ideas]
Hi @LyricTian! 👋 

Currently, I'm looking for an oauth2 solution that is able to be run in a serverless setup. This package comes up as one of the few on the oauth2 website but the package in itself is quite "server" oriented at the moment.

I was thinking and wanted to check whether you would be open to restructuring the package to facilitate multiple solutions to be possible? eg. a core that is less coupled to the original implementation which would allow for multiple solutions (eg. docker, AWS lambda, ...). 💡 

Would love to get your feedback on this and what any potential issues might be from your POV and whether you would be open to an MR to achieve this. 

Thanks! 🙇

**Comment by LyricTian** [comment]
That's a fantastic suggestion, I'll take it into consideration moving forward.  Regarding the specific plan for restructuring, I will organize my thoughts first and then share an update later on. Much appreciated.
